File: src/main/java/pattern/builder/Student.java
package pattern.builder;

import beans.User;


/**
 * Created by adimn on 2018/8/2.
 */
public class Student {
    private String name;
    private Integer age;
    private String Gender;

    private Student(StudentBuilder sb) {
        this.name = sb.name;
        this.age = sb.age;
        Gender = sb.gender;
    }

    public static class StudentBuilder{
        private String name = "ang";
        private Integer age = 10;
        private String gender = "female";
        public  StudentBuilder setName(String name){
            this.name = name;
            return this;
        }

        public StudentBuilder setAge(Integer age){
            this.age = age;
            return this;
        }

        public StudentBuilder setGender(String gender) {
            this.gender = gender;
            return this;
        }

        public Student build(){
            return new Student(this);
        }

    }

    public static void main(String[] args) {
        Student st = new Student.StudentBuilder().setName("ang").setAge(10).setGender("male").build();
    }

}

File: src/main/myscala/classconn/Student.scala
package classconn

import scala.beans.{BeanDescription, BeanProperty}

/**
  * Created by adimn on 2019/3/26.
  */
class Student {
  @BeanProperty var name = "ang"
//  def getName = name;
//  def setName(nn:String){ name = nn}
}

class Person(var name:String,var age:Int){
  private var hobby = "eat"
  println("whos your daddy!")
  def description = "name is "+name+", age is "+age

  class InnerPerson(var gender:String)

  def getInnerPerson(gender:String): InnerPerson ={
    name = "harry"
    println(name)
    print(description + ", gender is "+gender)
    println()
    new InnerPerson(gender)
  }

  //缺省构造函数
  def this(){
    this("Ang",10)
    Person.showInfo
    Person.num
  }
}

object Person{
  private var num:Long = 0;
  def showInfo{println( num)}
  def incNum(): Unit ={
    num +=1
  }
}

object StuConn{
  def main(args: Array[String]): Unit = {
//    var s1 = new Student
//    println(s1.getName)
//    s1.name = "potter"
//    println(s1.name)
//
//    s1.setName ( "hary")
//    println(s1.getName)
//
//    val p = new Person("Ang",10)
//    p.name = "super"
//    p.age = 100
//    println( p.description)
    val p1 = new Person("super",34)
//    p1.description

    var g1 = p1.getInnerPerson("female")
    Person.incNum()
    Person.showInfo
  }
}
